### PR TITLE
Added compression to HDFEventOutputer

### PR DESCRIPTION
**Pull request by Dr15Jones**
Also fixed a few bugs.
Still need to added compression algorithm name to file.

### CODE REVIEW

File: HDFEventOutputer.cc [hunk 1]
```diff
@@ -65,10 +67,10 @@ void HDFEventOutputer::setupForLane(unsigned int iLaneIndex, std::vector<DataPro
   }
   s.reserve(iDPs.size());
   offsetsAndBlob_.first.resize(iDPs.size()+1, 0);
-  if(iLaneIndex == 0) {
   for(auto const& dp: iDPs) {
     s.emplace_back(dp.name(), dp.classType());
   }
+  if(iLaneIndex == 0) {
```
Comment (Dr15Jones) [inline HDFEventOutputer.cc:73]: This fixes a bug where the code was crashing when using more than 1 lane.

File: HDFEventOutputer.cc [hunk 2]
```diff
@@ -122,7 +124,7 @@ void
 HDFEventOutputer::writeFileHeader(SerializeStrategy const& iSerializers) {
   constexpr hsize_t ndims = 1;
   constexpr hsize_t     dims[ndims] = {0};
-  hsize_t     chunk_dims[ndims] = {chunkSize_};
+  hsize_t     chunk_dims[ndims] = {static_cast<hsize_t>(chunkSize_)};
```
Comment (Dr15Jones) [inline HDFEventOutputer.cc:127]: This stops a compiler warning

File: HDFEventOutputer.cc [hunk 3]
```diff
@@ -168,10 +170,12 @@ std::pair<std::vector<uint32_t>, std::vector<char>> HDFEventOutputer::writeDataP
       auto offset = offsets[index++];
       std::copy(s.blob().begin(), s.blob().end(), buffer.begin()+offset );
     }
-    assert(buffer.size() == offset[index]);
+    assert(buffer.size() == offsets[index]);
```
Comment (Dr15Jones) [inline HDFEventOutputer.cc:173]: This fixes compilation under debug

**Comment by sabasehrish** [comment]
compression can be added as an attribute, I will update the HDFEventOutputer.

**Comment by Dr15Jones** [comment]
> compression can be added as an attribute, I will update the HDFEventOutputer.

Do you want to just do a pull request to the branch in my repo so it can go automatically into this pr?

**Comment by sabasehrish** [comment]
This one compressHDFEventOutputer ?

On Feb 22, 2022, at 4:33 PM, Chris Jones ***@***.******@***.***>> wrote:



compression can be added as an attribute, I will update the HDFEventOutputer.

Do you want to just do a pull request to the branch in my repo so it can go automatically into this pr?

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_hep-2Dcce2_root-5Fserialization_pull_28-23issuecomment-2D1048276140&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=fa7bLWMjbVL2ZoZcsm3m5_9lPdXvA1j7fgrBMEJikg0&m=MZb71yEJAtncm7cNI9noDvJ0WShx3-kPoTszbJLRWf1UXxq4KjyS8h8sFe9ArF1k&s=mLmlpIDCJ6pB5JASKSpRFonDGRlEPp_oS-9L0UoZch8&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ACR7ZN3CAB7MUCFAM5W5JHDU4QFMJANCNFSM5PCMHZVA&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=fa7bLWMjbVL2ZoZcsm3m5_9lPdXvA1j7fgrBMEJikg0&m=MZb71yEJAtncm7cNI9noDvJ0WShx3-kPoTszbJLRWf1UXxq4KjyS8h8sFe9ArF1k&s=rsnd4js0cz3PbkS8ghAbfsGk6cS81bT8dkxoiXogq_0&e=>.
Triage notifications on the go with GitHub Mobile for iOS<https://urldefense.proofpoint.com/v2/url?u=https-3A__apps.apple.com_app_apple-2Dstore_id1477376905-3Fct-3Dnotification-2Demail-26mt-3D8-26pt-3D524675&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=fa7bLWMjbVL2ZoZcsm3m5_9lPdXvA1j7fgrBMEJikg0&m=MZb71yEJAtncm7cNI9noDvJ0WShx3-kPoTszbJLRWf1UXxq4KjyS8h8sFe9ArF1k&s=eWBVbIDsPt8oAlWKeDW5fh0ZSWyJdzwrdDY81Toa5UA&e=> or Android<https://urldefense.proofpoint.com/v2/url?u=https-3A__play.google.com_store_apps_details-3Fid-3Dcom.github.android-26referrer-3Dutm-5Fcampaign-253Dnotification-2Demail-2526utm-5Fmedium-253Demail-2526utm-5Fsource-253Dgithub&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=fa7bLWMjbVL2ZoZcsm3m5_9lPdXvA1j7fgrBMEJikg0&m=MZb71yEJAtncm7cNI9noDvJ0WShx3-kPoTszbJLRWf1UXxq4KjyS8h8sFe9ArF1k&s=4qDqxV2pZMTsIrCukL1iG-9argKHEXJ7FsAm6_BUjww&e=>.

**Comment by Dr15Jones** [comment]
> This one compressHDFEventOutputer ?

Yes

**Pull request merged**